pkg/metrics/metrics.go

```diff
@@ -14,49 +14,59 @@
  * limitations under the License.                                           *
  ***************************************************************************/
 
-// Package event //
-package event
-
-// Metrics is the interface for event processor
-type Metrics interface {
-	SetQueueSize(queueSize int)
-	IncrSuccessFlushCount()
-	IncrFailFlushCount()
-	IncrRetryFlushCount()
+// Package metrics //
+package metrics
+
+import (
+	"sync"
+	"time"
+)
+
+// GenericMetrics provides the interface for the metrics
+type GenericMetrics interface {
+	Inc(key string)
+	Set(key string, val int64)
+	Get(key string) int64
 }
 
-// DefaultMetrics stores the actual metrics
-type DefaultMetrics struct {
-	QueueSize         int
-	SuccessFlushCount int64
-	FailFlushCount    int64
-	RetryFlushCount   int64
+// Metrics contains default metrics
+type Metrics struct {
+	startTime time.Time
+
+	metricsLock sync.RWMutex
+	metricsData map[string]int64
```

Comment:
FWIW, the `expvar.Map` wraps the `sync.Map`.

Reply:
ok, I have no preferences for locking map or using sync.Map.

Reply:
N/A